src/components/Slider/Slider.stories.tsx

```diff
@@ -40,3 +43,57 @@ export const Range: ComponentStory<typeof SliderComponent> = (args) => {
 		/>
 	)
 }
+
+const FlexRow = styled.div`
+	display: flex;
+`
+
+export const RangeWithInputs: ComponentStory<typeof SliderComponent> = (
+	args
+) => {
+	const [inputValue, setInputValue] = useState<SliderValue>([60, 80])
+	const [sliderValue, setSliderValue] = useState<SliderValue>([60, 80])
+
+	const startValue = (inputValue as (string | number)[])[0]
+	const endValue = (inputValue as (string | number)[])[1]
```

Comment:
What are these type machinations about?  Why do we need to force the type?

Reply:
`inputValue` is of type `SliderValue` which is defined as `string | number | (string | number)[]`.

If I do just

```
const startValue = inputValue[0]
const endValue = inputValue[1]
```

if that's what you meant, the compiler complains:

> Element implicitly has an 'any' type because expression of type '0' can't be used to index type 'SliderValue'.
>   Property '0' does not exist on type 'SliderValue'.

I can do something like this

```	
const [inputValue, setInputValue] = useState<SliderValue>(['60', '80'])
const [sliderValue, setSliderValue] = useState<SliderValue>(['60', '80'])

const startValue = (inputValue as string[])[0]
const endValue = (inputValue as string[])[1]
```

But setting the value as string seems to me so unintuitive. 

Or, I can force the type `as number[]` but then I'd have to cast it on all places where I set the value to number because the noUiSlider component works apparently internally with strings and that's what I'm getting.

Or do you have any suggestions?

Reply:
Will also developers using this component have to do the same force-typing? Maybe by giving so many options we over-complicate things - IDK, just a thought 🤷‍♂️ . If you know what I mean. 🤔

Reply:
I know what you mean. Well, in scenario where on top of the slider, there will be text inputs like in this case, they would have to do exactly this. You're right, that's not ideal. Although I created this story specifically for the case of showing how to use it.

The problem is that the component can be used both as single slider and as multi slider so the value type consists of both scalar types (`string | number`) and of array (`(string | number)[]`). Exactly because of these kinds of issues, we have so far for all other components different components for single value and multi value use case (Select vs MultiSelect, DatePicker vs DateRangePicker).

In this component, it seemed to me less problematic but I can imagine splitting it up into Slider and ... RangeSlider or MultiSlider. In Blueprint, they use for example three separate components - Slider (1 handle), RangeSlider (2 handles), MultiSlider (n handles).

So maybe splitting the component would be the best solution.

Reply:
Agree, let's split it for sake of better DX. 🙏 🙂